src/aspire/source/coordinates.py

```diff
@@ -250,6 +250,203 @@ def _get_mrc_shapes(self):
 
         return mrc_shapes
 
+    def import_ctf(self, ctf):
+        """
+        Given a RELION CTF STAR file, or list of CTF STAR files--one for each micrograph--, this function adds
+        corresponding CTFFilter objects to the `CoordinateSource`'s unique
+        filter list, and also populates the metadata.
+        """
+        ctf_cols = [
+            "_rlnVoltage",
+            "_rlnDefocusU",
+            "_rlnDefocusV",
+            "_rlnDefocusAngle",
+            "_rlnSphericalAberration",
+            "_rlnAmplitudeContrast",
+        ]
+
+        # attributes to be populated by the different CTF's
+        # unique CTFFilter objects
+        filters = []
```

Comment:
This is because for each filter, the corresponding indices in each of these arrays are updated all together.

So for instance, if the 7th CTF filter corresponds to indices 70 through 100, those indices need to already exist. That's why I decided on creating the empty arrays in advance

Reply:
Can't link it but see e.g. lines 376-377 in this file

Reply:
The only issue I see there is potentially cluttering the `__init__`. What do you think?